broker/src/main/java/io/camunda/zeebe/broker/Broker.java

```diff
@@ -446,15 +436,38 @@ public PartitionManager getPartitionManager() {
     private AtomixCluster atomix;
   }
 
-  private static class BrokerStartupActor extends Actor {
+  /**
+   * Temporary helper object. This object is needed during the transition of broker startup/shutdown
+   * steps to the new concept. Afterwards, the expectation is that this object will merge with the
+   * Broker and that then the Broker becomes the actor
+   */
+  private static final class BrokerStartupActor extends Actor {
+
+    private final BrokerStartupProcess brokerStartupProcess;
+
+    private final int nodeId;
+
+    private BrokerStartupActor(final BrokerStartupContextImpl startupContext) {
+      nodeId = startupContext.getBrokerInfo().getNodeId();
+      startupContext.setConcurrencyControl(actor);
+      brokerStartupProcess = new BrokerStartupProcess(startupContext);
+    }
 
     @Override
     public String getName() {
-      return "Broker Startup Actor";
+      return buildActorName(nodeId, "Startup");
+    }
+
+    private ActorFuture<BrokerContext> start() {
+      final ActorFuture<BrokerContext> result = createFuture();
+      actor.call(() -> actor.runOnCompletion(brokerStartupProcess.start(), result));
```

Comment:
```suggestion
      actor.submit(() -> actor.runOnCompletion(brokerStartupProcess.start(), result));
```

:question: Or is there a reason for using call specifically? I know `call` and `run` both prepend tasks instead of appending them to the task queue, but I don't see a need for it here, unless I'm missing something.

Reply:
If I recall correctly, submit must be called from an Actor thread, but call can be called from anywhere. But honestly, this whole actor stuff is black magic to me. 

So please advise which is the right method to call from outside an actor thread and appending tasks to the end of the queue.

Reply:
You're correct - it's also confusing for me :smile: So, it looks like `run`, if called from within an actor thread, will prepend jobs, but if called outside will append jobs (i.e. behave like `submit`). That's...confusing, to say the least. I would still propose `run` since we don't want a return value.